WordPress/src/test/java/org/wordpress/android/ui/mysite/ScanAndBackupSourceTest.kt

```diff
@@ -109,8 +111,51 @@ class ScanAndBackupSourceTest : BaseUnitTest() {
 
         val result = scanAndBackupSource.buildSource(siteId).take(2).toList().last()
 
-        assertThat(result.backupAvailable).isFalse()
-        assertThat(result.scanAvailable).isFalse()
+        assertThat(result.backupAvailable).isFalse
+        assertThat(result.scanAvailable).isFalse
+    }
+
+    @Test
+    fun `Scan not visible on wpcom sites even when Scan product is available`() = test {
+        init(scanScreenFeatureEnabled = true)
+        whenever(site.siteId).thenReturn(siteRemoteId)
+        whenever(jetpackCapabilitiesUseCase.getJetpackPurchasedProducts(siteRemoteId)).thenReturn(
+                JetpackPurchasedProducts(scan = true, backup = false)
+        )
+        whenever(site.isWPCom).thenReturn(true)
+
+        val result = scanAndBackupSource.buildSource(siteId).take(2).toList().last()
+
+        assertThat(result.scanAvailable).isFalse
+    }
+
+    @Test
+    fun `Scan not visible on atomic sites even when Scan product is available`() = test {
+        init(scanScreenFeatureEnabled = true)
+        whenever(site.siteId).thenReturn(siteRemoteId)
+        whenever(jetpackCapabilitiesUseCase.getJetpackPurchasedProducts(siteRemoteId)).thenReturn(
+                JetpackPurchasedProducts(scan = true, backup = false)
+        )
+        whenever(site.isWPComAtomic).thenReturn(true)
+
+        val result = scanAndBackupSource.buildSource(siteId).take(2).toList().last()
+
+        assertThat(result.scanAvailable).isFalse
+    }
+
+    @Test
+    fun `Scan visible on non-wpcom sites when Scan product is available and feature flag enabled`() = test {
+        init(scanScreenFeatureEnabled = true)
+        whenever(site.siteId).thenReturn(siteRemoteId)
+        whenever(jetpackCapabilitiesUseCase.getJetpackPurchasedProducts(siteRemoteId)).thenReturn(
+                JetpackPurchasedProducts(scan = true, backup = false)
+        )
+        whenever(site.isWPCom).thenReturn(false)
+        whenever(site.isWPComAtomic).thenReturn(false)
```

Comment:
Suggestion (💡): Since you explicitly added the `.thenReturn(false)` here, for both cases (`isWPCom` and `isWPComAtomic`), despite the fact that this is being setup again during `setUp()`, which I like, I suggest adding the missing `.thenReturn(false)` case on the other two cases above, to make that explicit as well.

Reply:
Hmm, I usually explicitly list only things which are relevant for test/what the tests is trying to verify.

"Scan not visible on atomic sites even when Scan product is available" is trying to verify Atomic sites, so only "whenever(site.isWPComAtomic).thenReturn(true)" is listed.

"Scan not visible on wpcom sites even when Scan product is available" is trying to verify wpcom sites, so only "whenever(site.isWPCom).thenReturn(true)" is listed.

"Scan visible on non-wpcom sites when Scan product is available and feature flag enabled`" is trying to verify non-wpcom sites, which are defined as !wpcom && !atomic -> therefore 
"whenever(site.isWPCom).thenReturn(false)
        whenever(site.isWPComAtomic).thenReturn(false)" is listed.
        
        
What do you think about this approach? Would you still prefer if it was explicitly listed everywhere?

Reply:
👋 @malinajirka !

Yeap, yeap, I got that, thanks for sharing your process. I would just add them explicitly just to show that one of them is `false` since this bit of information matters in that test context and the last test contain both as well. But, again, this is very opinionated everyone would do it differently, I just wanted to mention what I would have done.

PS: This is very minor so please ignore as such.

Reply:
Updated in 078f00e2583508fd0b93227b89e719c5863eedba

Reply:
So bad news - the tests were complaining about "unnecessary stubbing" so I had to revert this change 😞 .